### ISSUE TITLE
InfluxDB Datasource: Undefined identifier RETENTION and POLICIES after upgrading to Grafana 9.5.x

### ISSUE THREAD

**Issue by ikogan** [grafana/grafana#67610]
### 
# What went wrong?

**What happened**:

- Upgraded to Grafana 9.5.1 from 9.4.x.
- Have InfluxDB plugin configured to use Flux queries
- Lots of error logs that look like this begin to appear:
  ```
  logger=tsdb.influx_flux t=2023-05-02T02:34:47.392159245Z level=warn msg="Flux query failed" err="invalid: error @1:1-1:5: undefined identifier SHOW\n\nerror @1:6-1:15: undefined identifier RETENTION\n\nerror @1:16-1:24: undefined identifier POLICIES\n\nerror @1:25-1:27: undefined identifier on" query="SHOW RETENTION POLICIES on \"undefined\""
  ```
- There doesn't seem to be any visible issues when actually using Grafana

**What did you expect to happen**:

  - No repeated constant errors.

### 
# How do we reproduce it?

I'm not 100% sure how to reproduce this but I believe the issue is that [this change](https://github.com/grafana/grafana/pull/66466) is applying to datasources using Flux on InfluxDB 2.x when it should only apply to InfluxQL on 1.x. If that's the case, the following steps should work:

**Step 1**: Setup an InfluxDB 2.x instance. I'm currently using 2.7.0.
**Step 2**:  Setup Influx DB datasource using Flux as the query language.
**Step 3**: Setup some dashboards from data in the datasource.
**Step 4**: Observe errors in the log about retention policies.

### 
# What Grafana version are you using?

v9.5.1

### 
### Is the bug inside a Dashboard Panel?

No, unless being in a query that a dashboard panel uses counts.

### Grafana Platform?

Kubernetes

### User's OS?

Ubuntu 22.04

### User's Browser?

Chrome Version 112.0.5615.165 (Official Build) (64-bit)

### Is this a Regression?

Yes

### Are Datasources involved?

InfluxDB 5.0.0

**Comment by itsmylife** [comment]
@ikogan I am able to reproduce this issue. Thanks for reporting this. It is clearly a bug and as you stated we should not run a retention policy query for flux. I will make the necessary change.

**Comment by ollie-campbell** [comment]
Hi, Just wondering how I can find out when this fix might be publicly release? Also seeing this issue.

**Comment by itsmylife** [comment]
@ollie-campbell @ikogan I checked that again and it will be available in `v10.0.0`. 
See `v10.0.0-preview`: https://grafana.com/grafana/download/10.0.0-preview?pg=get&plcmt=selfmanaged-box1-cta1
`v9.5.x` will only receive security fixes and critical bug fixes.

**Comment by ollie-campbell** [comment]
> @ollie-campbell @ikogan I checked that again and it will be available in `v10.0.0`. See `v10.0.0-preview`: https://grafana.com/grafana/download/10.0.0-preview?pg=get&plcmt=selfmanaged-box1-cta1 `v9.5.x` will only receive security fixes and critical bug fixes.

Ok great. I presume v10's delivery date is very much down to how many issues are found in testing? But, any idea how long it might be?

**Comment by itsmylife** [comment]
@ollie-campbell I cannot give an exact date but it is close. In GrafanaCon we will give more information about it 
See the blog post: https://grafana.com/blog/2023/05/30/grafana-labs-oss-projects-discover-the-latest-and-greatest-features-and-updates/
If you'd like you can register for the session https://grafana.com/about/events/grafanacon/2023/session/grafana-10-deep-dive/?pg=blog&plcmt=body-txt

**Comment by ollie-campbell** [comment]
Hi @itsmylife.
I've just updated to v10 now that its released and am still seeing the same issues.

**Comment by itsmylife** [comment]
@ollie-campbell v10 had some code freeze even before I merge the fix (I thought I merged it before the freeze 🤦 ). `v10.0.1` will have the fix. I am sorry that I wasn't clear enough earlier.  On the 22nd of June, there will be a release.

**Comment by ollie-campbell** [comment]
I can confirm v10.0.1 stops this error appearing.

**Comment by TimfruPD** [comment]
Hi, I also updated to v10.0.1 (5a30620b85) running on Windows, but still get the error in the logs: 
`logger=tsdb.influx_flux t=2023-07-10T08:33:11.0171376+02:00 level=warn msg="Flux query failed" err="invalid: error @1:1-1:5: undefined identifier SHOW\n\nerror @1:6-1:15: undefined identifier RETENTION\n\nerror @1:16-1:24: undefined identifier POLICIES\n\nerror @1:25-1:27: undefined identifier on" query="SHOW RETENTION POLICIES on \"undefined\""
`

**Comment by itsmylife** [comment]
@TimfruPD I double-checked the files and the code is there. Could you please (I am sure you did it but still wanna suggest it) try cleaning the browser cache? Also, you can try to upgrade to `v10.0.2`. Let me know if that help.

**Comment by TimfruPD** [comment]
@itsmylife Hi, it seems to have resolved itself after some days (v10.0.1). I think indeed one of the clients browsers cache was still causing the error to show up. Thank you for fixing this error!

**Comment by itsmylife** [comment]
@nrukavkov I tested once more and it was fixed. Could you please double-check that the browser cache is cleaned?

**Comment by tejasharwin** [comment]
@itsmylife Hey ismail, 

I've detailed a similar issue for InfluxQL here on community forums:
https://community.grafana.com/t/redundant-queries-made-show-retention-policies/104150

Essentially, whenever we open our dashboard a redundant query is being made:
`SHOW RETENTION POLICIES on “inventory_management"`

Could you please advise why this is happening and what the resolution is?

**Comment by tejasharwin** [comment]
Hi Team, any idea on the above please?

**Comment by itsmylife** [comment]
Hi @tejasharwin I will take a look at that post today. Thanks for pointing it out.

**Comment by itsmylife** [comment]
@tejasharwin I want to share the same response here too:

Hi @ajetsharwin It is happening because of this lines https://github.com/grafana/grafana/blob/v9.5.x/public/app/plugins/datasource/influxdb/datasource.ts#L193-L214

Basically, for that version, I changed the code to load the retention policies in the beginning so I can show the "default" retention policy on the panel if there is none selected. Then I changed that behavior again. Not because it was wrong but I need to support the new influxdb engine. See here: https://github.com/grafana/grafana/pull/72467

So there is nothing wrong with having that query being made.
